File: go/number-guessing-game/internal/difficulty/difficulty.go
package difficulty

import "fmt"

func SelectingDifficulty() (attempts int) {
	fmt.Println("Select difficulty level:")
	fmt.Println("1. Easy (10 chances)")
	fmt.Println("2. Medium (5 chances)")
	fmt.Println("3. Hard (3 chances)")

	var choice int
	fmt.Scanf("%d", &choice)

	switch choice {
	case 1:
		attempts = 10
	case 2:
		attempts = 5
	case 3:
		attempts = 3
	default:
		fmt.Println("Invalid choice. Defaulting to Medium difficulty.")
		attempts = 5
	}

	fmt.Printf("You have %d chances to guess the number.\n", attempts)

	return attempts
}

File: python/number-guessing-game/difficulty.py
def select_difficulty():
    print("Select difficulty level:")
    print("1. Easy (10 chances)")
    print("2. Medium (5 chances)")
    print("3. Hard (3 chances)")

    while True:
        try:
            choice = int(input("Enter your choice: "))
            if choice == 1:
                return 10
            elif choice == 2:
                return 5
            elif choice == 3:
                return 3
            else:
                print("Invalid choice. Please enter 1, 2, or 3.")
        except ValueError:
            print("Invalid input. Please enter a number.")
